### DISCUSSION TITLE
Encapsulated plugins, a discussion

### DISCUSSION THREAD

**Discussion by scottcwilson** [zencart/zencart#5534] [Ideas]
One negative side effect of the structure of the encapsulated plugins is that prior versions are not easily compared, since they are in a different directory hierarchy.

One possibility would be to have a single hierarchy and a file containing the version number.

**Comment by lat9** [comment]
IMO, it's more than version handling.  Having (finally) figured out to to a comparison of @mc12345678's change to the DisplayLogs plugin (#5515), I also noticed that there's no 'change history' associated with a plugin's versioning.

My local zc158 was still running v3.0.0 of DisplayLogs, so I was presented with two radio buttons for my upgrade: v3.0.1 and v3.0.2 ... not particularly informative!  It would be 'nice' to know what features/bugfixes the particular version included.

Perhaps we should also include a minimum PHP supported and a maximum PHP version tested?  Otherwise, a plugin designed for zc158/PHP 8.2 could be installed on a zc158/PHP 7.3 site and fail miserably.

**Comment by lat9** [comment]
I'll note also that a plugin's `manifest.php` contains a `changelog` and `gethub_repo` value, neither of which are used by the plugin sub-system.

While I'm at it, what information is the **Test** column in the **Plugin Manager** supposed to convey?
![image](https://user-images.githubusercontent.com/2685585/212543830-35585396-bfd9-4e57-9d21-39de738dfc92.png)

**Reply by lat9** [reply]
I'll note that the "Test" column doesn't display in zc157.

**Comment by scottcwilson** [comment]
@zcwilt Can you answer this?

**Comment by lat9** [comment]
What we could explore (for the 'core' plugin(s)) is to use `git rename` to rename to plugin's version directory to reflect its new one.  That way the plugin's file-set would retain is change-related 'juice' on an update.
